ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHModelServing.resource

```diff
@@ -261,44 +261,76 @@ Get Model Inference
     [Documentation]    Returns the inference result after sending a POST request to a deployed
     ...    model endpoint. If token authentication is needed for the model, ${token_auth} should be
     ...    set to ${TRUE}.
-    [Arguments]    ${model_name}    ${inference_input}    ${token_auth}=${FALSE}
-    ...    ${project_title}=${NONE}
+    [Arguments]    ${model_name}    ${inference_input}    ${token_auth}=${FALSE}   ${project_title}=${NONE}
+    ...    ${kserve_mode}=Serverless   ${deployment_mode}='UI'   ${service_port}=8888   ${end_point}=${NONE}
+    ${curl_cmd}=     Set Variable    ${NONE}
     ${self_managed}=    Is RHODS Self-Managed
-    ${url}=    Get Model Route Via UI    ${model_name}
-    ${kserve}=    Run Keyword And Return Status    SeleniumLibrary.Page Should Contain    Single-model serving enabled
-    ${curl_cmd}=     Set Variable    curl -s ${url} -d ${inference_input}
-    IF    ${token_auth}
-        IF    "${project_title}" == "${NONE}"
-            ${project_title}=    Get Model Project    ${model_name}
+    IF  ${deployment_mode} == 'UI'
+        ${url}=    Get Model Route Via UI    ${model_name}
+        ${kserve}=    Run Keyword And Return Status    SeleniumLibrary.Page Should Contain    Single-model serving enabled
+        ${curl_cmd}=     Set Variable    curl -s ${url} -d ${inference_input}
+        IF    ${token_auth}
+            IF    "${project_title}" == "${NONE}"
+                ${project_title}=    Get Model Project    ${model_name}
+            END
+            ${token}=    Get Access Token via UI    ${project_title}
+            ${curl_cmd}=     Catenate    ${curl_cmd}    -H "Authorization: Bearer ${token}"
+        END
+        IF  ${kserve}
+            Fetch Knative CA Certificate    filename=openshift_ca_istio_knative.crt
+            ${curl_cmd}=     Catenate    ${curl_cmd}    --cacert openshift_ca_istio_knative.crt
+        ELSE IF  ${self_managed}
+            Fetch Openshift CA Bundle
+            ${curl_cmd}=     Catenate    ${curl_cmd}    --cacert openshift_ca.crt
         END
-        ${token}=    Get Access Token Via UI    ${project_title}
-        ${curl_cmd}=     Catenate    ${curl_cmd}    -H "Authorization: Bearer ${token}"
     END
-    IF  ${kserve}
-        Fetch Knative CA Certificate    filename=openshift_ca_istio_knative.crt
-        ${curl_cmd}=     Catenate    ${curl_cmd}    --cacert openshift_ca_istio_knative.crt
-    ELSE IF  ${self_managed}
-        Fetch Openshift CA Bundle
-        ${curl_cmd}=     Catenate    ${curl_cmd}    --cacert openshift_ca.crt
+    IF  ${deployment_mode} == 'Cli'
+        ${rc}   ${cmd_op}   Run And Return Rc And Output    oc get isvc ${model_name} -o jsonpath='{.metadata.name}'
+        Should Be Equal As Integers    ${rc}    0
+         IF  "${cmd_op}" == "${model_name}"
```

Comment:
This doesn't make sense, you're returning the model name from the ISVC and then checking that it's equal to the model name, it will always be true.
This condition will be true also when deploying models with modelmesh. 

You can achieve what you're looking for here with something like 
```
oc get isvc {model_name} -o jsonpath='{.metadata.annotations.serving\.kserve\.io/deploymentMode}' -n {my_ns}
```
It will return `ModelMesh` if it's using modelmesh or an empty string otherwise.

Reply:
thanks i did not think of modelmesh thing ill apply it